### PR TITLE
ES2015ify – Part 2

### PR DESCRIPTION
**Pull request by schnittstabil**
ES2015ify – Part 2:
* Drop support for Node v0.10 and v0.12
* Upgrade `cp-file` to `^v4.0`
* Remove `pinkie-promise` and `object-assign`

Additionally:
* Fix globby error handling
* Increase code coverage to 100%

Closes: #36

### CODE REVIEW

File: cpy-error.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+class CpyError extends NestedError {
+	constructor(message, nested) {
+		super(message, nested);
+		Object.assign(this, nested);
+		this.name = 'CpyError';
+	}
+}
+
+module.exports = CpyError;
```
Comment (sindresorhus) [inline cpy-error.js]: You need to add this file to `files` in package.json

File: index.js [hunk 2]
```diff
@@ -38,29 +20,26 @@ function preprocessDestPath(srcPath, dest, opts) {
...
+		.catch(err => {
 			throw new CpyError('Cannot glob `' + src + '`: ' + err.message, err);
-		});
+		})
+		.then(files => Promise.all(files.map(srcPath => {
+			const from = preprocessSrcPath(srcPath, opts);
+			const to = preprocessDestPath(srcPath, dest, opts);
+
+			return cpFile(from, to, opts).catch(err => {
+				throw new CpyError('Cannot copy from `' + from + '` to `' + to + '`: ' + err.message, err);
```
Comment (sindresorhus) [inline index.js]: Use template literal

File: index.js [hunk 3]
```diff
@@ -38,29 +20,26 @@ function preprocessDestPath(srcPath, dest, opts) {
...
-				var to = preprocessDestPath(srcPath, dest, opts);
-
-				return cpFile(from, to, opts).catch(function (err) {
-					throw new CpyError('Cannot copy from `' + from + '` to `' + to + '`: ' + err.message, err);
-				});
-			}));
-		})
-		.catch(function (err) {
+		.catch(err => {
 			throw new CpyError('Cannot glob `' + src + '`: ' + err.message, err);
```
Comment (sindresorhus) [inline index.js]: Template literal

File: cpy-error.js (hunk 1 above)
Comment (schnittstabil) [inline cpy-error.js]: Damn! Same mistake within one week :cry:

**Comment by sindresorhus** [comment]
Awesome! 

Anything else you'd like to land (especially breaking changes) before I do a major release?

**Comment by schnittstabil** [comment]
Recursively copying and/or progress reporting, but both need further investigations and I'm not sure if I find enough time this weekend. If you want to, you may postpone the next major until monday.

**Comment by sindresorhus** [comment]
Sure, no rush with a new major release.

**Pull request merged**